src/worker/tasks/sendTransactionWorker.ts

```diff
@@ -78,17 +78,15 @@ const handler: Processor<string, void, string> = async (job: Job<string>) => {
   // For example, the developer retried all failed transactions during an RPC outage.
   // An errored queued transaction (resendCount = 0) is safe to retry: the transaction wasn't sent to RPC.
   if (transaction.status === "errored" && resendCount === 0) {
+    const { errorMessage, ...omitted } = transaction;
     transaction = {
-      ...{
-        ...transaction,
-        nonce: undefined,
```

Comment:
This was the bug. By setting values to undefined, they were _explicitly_ set to undefined in the type. This is unexpected because the typing implies undefined is not a valid value, which leads to weird edge cases like the code not handling undefined values.

Reply:
The fix here was to flatten this code to override certain fields only.

Note: 
- nonce doesn't need to be unset since we're calling `_sendTransaction()` again which generates a new nonce as expected.
- gas fees don't need to be unset since overrides were moved to a separate nested object.